src/hotspot/share/memory/universe.cpp

```diff
@@ -652,28 +676,28 @@ void Universe::create_preallocated_out_of_memory_errors(TRAPS) {
   }
   _out_of_memory_errors = OopHandle(vm_global(), oom_array());
 
-  Handle msg = java_lang_String::create_from_str("Java heap space", CHECK);
+  Handle msg = java_lang_String::create_from_str(MSG_JAVA_HEAP, CHECK);
   java_lang_Throwable::set_message(oom_array->obj_at(_oom_java_heap), msg());
 
-  msg = java_lang_String::create_from_str("C heap space", CHECK);
+  msg = java_lang_String::create_from_str(MSG_C_HEAP, CHECK);
   java_lang_Throwable::set_message(oom_array->obj_at(_oom_c_heap), msg());
 
-  msg = java_lang_String::create_from_str("Metaspace", CHECK);
+  msg = java_lang_String::create_from_str(MSG_METASPACE, CHECK);
   java_lang_Throwable::set_message(oom_array->obj_at(_oom_metaspace), msg());
 
-  msg = java_lang_String::create_from_str("Compressed class space", CHECK);
+  msg = java_lang_String::create_from_str(MSG_CLASS_METASPACE, CHECK);
```

Comment:
I meant intern these two strings here and check for equality?

Reply:
Sorry Coleen I do not understand what you mean by this. How do you compare the "interned" oop String with the incoming `char*` text?

Reply:
OopHandle Universe::_metaspace_string;
...
msg = String::intern("Metaspace", CHECK);
java_lang_Throwable::set_message(oom_array->obj_at(_oom_metaspace, msg());
Universe::_metaspace_string = OopHandle(vm_global(), msg);
...
bool Universe::is_metaspace_oome(oop exc) {
  return java_lang_Throwable::get_message(exc) == _metaspace_string.resolve();
}

Reply:
Maybe not.  Maybe the exception you're checking could be newly allocated and not use an interned string.

Reply:
except the things you're checking for are not.  This code could also increment the counter.
  if (out_of_compressed_class_space) {
    THROW_OOP(Universe::out_of_memory_error_class_metaspace());
  } else {
    THROW_OOP(Universe::out_of_memory_error_metaspace());
  }

Reply:
sorry I deleted my last comment.

Reply:
I'm getting completely lost on the suggestions now sorry. The logic of this code is really convoluted.

Reply:
Yes, I agree. I figured it out once and had to start from scratch to try to figure it out again.

Reply:
Like this: https://github.com/openjdk/jdk/pull/6901

Reply:
Okay - yes that looks reasonably simple. Thanks.

Reply:
@coleenp @dholmes-ora  Thanks for commenting on this. I make some changes according to review feedback.